### ISSUE TITLE
[Bug]: Vacuum full deleted the cache tables under _timescaledb_catalog

### ISSUE THREAD

**Issue by ngosala** [timescale/timescaledb#4514]
### What type of bug is this?

Crash

### What subsystems and features are affected?

Distributed hypertable

### What happened?

Vacuum full deleted the cache tables under _timescaledb_catalog. I can no longer perform any valid queries on the db. Is there a way to generate this cache ? 
ALTER EXTENSION timescaledb UPDATE; to 2.7.1 doesn't work 
And DROP EXTENSION doesn't work as well. 
I don't want to lose 30+TB of my data. Is there any fix to recover my db ? 

### TimescaleDB version affected

2.3.1

### PostgreSQL version used

12

### What operating system did you use?

Centos

### What installation method did you use?

RPM

### What platform did you run on?

On prem/Self-hosted

### Relevant log output and stack trace

```bash
2022-07-17 02:30:09.668 UTC [283795] ERROR:  OID lookup failed for table "_timescaledb_catalog.hypertable"
2022-07-17 02:30:09.668 UTC [283795] STATEMENT:  SELECT c.oid
	FROM pg_catalog.pg_class c
	     LEFT JOIN pg_catalog.pg_namespace n
	     ON n.oid OPERATOR(pg_catalog.=) c.relnamespace
	WHERE c.relkind OPERATOR(pg_catalog.=) ANY
	    (array['r', 'S', 'v', 'm', 'f', 'p'])
	  AND c.relname OPERATOR(pg_catalog.~) '^(user_auth)$' COLLATE pg_catalog.default
	  AND n.nspname OPERATOR(pg_catalog.~) '^(public)$' COLLATE pg_catalog.default
```

### How can we reproduce the bug?

```bash
by running vacuum full on the system
```

**Comment by svenklemm** [comment]
Are you sure you are connected to the right database. VACUUM FULL should not delete any extension tables and postgres will prevent deleting extension tables by default.

**Comment by ngosala** [comment]
@svenklemm 
Thanks for looking into this. This is a bug in the previous versions of timescale, which is fixed in 2.6.1 
ref: https://github.com/timescale/timescaledb/issues/3924
Vacuum full task deleted everything under _timescale_catalog schema and hence, I couldn't query any data that I had under my db. 
I tried updating timescale to see if the `ALTER EXTENSION timescaledb UPDATE;` would restore/regen cache but this command failed with oid errors 
`DROP extension` used to hang forever 
`CREATE extension` failed as extension already exists. 
`REINDEX` was failing with oid errors too

I tried to manually gen tables but the whole system was in a state of deadlock preventing me from creating anything new and accessing what's already there. including _timescale_catalog schema

How I restored my DB:
Looking at the above threads, the problem seems to arise from max_parallell_workers and I disabled them. Then I disabled timescale as shared_preload_libraries 
`shared_preload_libraries = 'timescaledb'` as the problem is just with timescale catalog. 
restarted postgres and then I ran 
`REINDEX SYSTEM <dbname>` which re-generated _timescale_catalog and then I restored the settings to before
